controller/eks-cluster-config-handler.go

```diff
@@ -1152,6 +1152,11 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *eksv1.EKSClusterConfi
 			updateNodegroupProperties = true
 			_, err := eksService.UpdateNodegroupVersion(ngVersionInput)
 			if err != nil {
+				if version, ok := templateVersionsToAdd[aws.StringValue(ng.NodegroupName)]; ok {
+					// If there was an error updating the node group and a Rancher-managed launch template version was created,
+					// then the version that caused the issue needs to be delete to prevent bad versions from piling up.
+					deleteLaunchTemplateVersions(config.Status.ManagedLaunchTemplateID, []*string{aws.String(version)}, ec2Service)
+				}
 				return config, err
```

Comment:
This made me realize something, isn't it possible for config's status to be change in `newLaunchTemplateVersionIfNeeded` but never actually updated? In this case we wouldn't want it updated any way I think, but I'm guessing this messes up the cache. Also, seem like a side effect of the function that should maybe be brought out so it's easier to track.

Reply:
Yes, you're right. That absolutely needs to be pulled out. Will update.